### ISSUE TITLE
A11y / Course Filter / Learn - Toggle to open filters does not have a useful accessible name

### ISSUE THREAD

**Issue by joedolson** [WordPress/Learn#2784]
## Type of feedback
//dev

## Description

The toggle that opens the course status filters is labeled using the currently selected status. Buttons to perform an action should be labeled using information about what the control does. Having the current status visible is OK; but it shouldn't be the only content.

This would be better labeled using something like "Filter Courses: All" or even "Filter: All". This would inform the user what they can do with the control. 

At https://learn.wordpress.org/my-courses/

## Screenshots or screen recording (optional)

![image](https://github.com/user-attachments/assets/b26cb63d-52a3-4e7d-8c85-87699af40d16)

**Comment by ryelle** [comment]
Just noting that this is [the shared "Filter" component](https://github.com/WordPress/wporg-mu-plugins/tree/trunk/mu-plugins/blocks/query-filter) and we use this pattern across wp.org (like [Patterns](https://wordpress.org/patterns/)). The "single select" filters use the currently-selected item as the button text. It had been using the action in the label (ex, "Sort: Newest" or "Filter: Active"), but it was dropped in the design here https://github.com/WordPress/pattern-directory/issues/648#issuecomment-2031738949.

I'm pretty sure it can be fixed here without requiring a larger change, but if this pattern doesn't work on Learn, it probably doesn't work on the other sites.

**Comment by joedolson** [comment]
Well, I'd call that an accessibility regression since I'd last reviewed the filter component; it makes it harder to understand the control. The minimal labeling is probably fine for people who use the site all the time, but isn't very welcoming to new users.  

Yes, I agree that this is going to be the same problem everywhere this component is used.
